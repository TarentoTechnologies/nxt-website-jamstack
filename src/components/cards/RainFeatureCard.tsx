import { GatsbyImage, getImage } from "gatsby-plugin-image";
import React from "react";

import { carbon } from "../../styles/style-guide/ColorGuide.module.css";
import { cardHeader } from "../../styles/style-guide/Typography.module.css";
import {
  headingStyles,
  imgStyles,
  subTextStyles,
} from "./RainFeatureCard.module.css";

interface RainFeatureCardProps {
  heading: string;
  subText: string;
  img: any;
  isImage: boolean;
}

export const RainFeatureCard = ({
  heading,
  subText,
  img,
  isImage,
}: RainFeatureCardProps) => {
  return (
    <div data-testid="rain-feature-card">
      <div className={`${imgStyles}`}>
        {!isImage ? (
          <GatsbyImage
            image={
              getImage(img?.localFile)
                ? getImage(img?.localFile)
                : img?.localFile
            }
            alt={""}
            data-aos="fade-up"
          />
        ) : (
          <img
            src={img}
            alt={heading}
            title={heading}
            className={`img-fluid`}
            data-aos="fade-up"
          />
        )}
      </div>
      <div
        className={`${headingStyles} ${cardHeader} ${carbon}`}
        data-aos="fade-up"
      >
        {heading}
      </div>
      <p className={`${subTextStyles} ${carbon}`} data-aos="fade-up">
        {subText}
      </p>
    </div>
  );
};
