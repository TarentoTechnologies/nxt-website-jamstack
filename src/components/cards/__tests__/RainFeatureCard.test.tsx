import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { RainFeatureCard } from "../RainFeatureCard";

const imgData = {
  localFile: {
    childImageSharp: {
      gatsbyImageData: {
        layout: "constrained",
        placeholder: {
          fallback:
            "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAJABQDASIAAhEBAxEB/8QAFwAAAwEAAAAAAAAAAAAAAAAAAAIDBf/EABUBAQEAAAAAAAAAAAAAAAAAAAAB/9oADAMBAAIQAxAAAAHLWq1MoH//xAAZEAACAwEAAAAAAAAAAAAAAAAAEQEQITH/2gAIAQEAAQUCWaIjlf/EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQMBAT8BP//EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQIBAT8BP//EABQQAQAAAAAAAAAAAAAAAAAAACD/2gAIAQEABj8CX//EABsQAAEEAwAAAAAAAAAAAAAAAAABEBFxMUFR/9oACAEBAAE/IdiTNC2JcMNP/9oADAMBAAIAAwAAABAwD//EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQMBAT8QP//EABcRAQADAAAAAAAAAAAAAAAAAAABESH/2gAIAQIBAT8QnVP/xAAcEAEAAQQDAAAAAAAAAAAAAAABEQAQITFhkbH/2gAIAQEAAT8QJlFDUvaDwnauVbHdv//Z",
        },
        images: {
          fallback: {
            src: "/static/c5ab777b81ede6691c6b6d65988b39cb/02c3d/features_1_6765d2890b.png",
            srcSet:
              "/static/c5ab777b81ede6691c6b6d65988b39cb/07064/features_1_6765d2890b.png 193w,\n/static/c5ab777b81ede6691c6b6d65988b39cb/35e06/features_1_6765d2890b.png 386w,\n/static/c5ab777b81ede6691c6b6d65988b39cb/02c3d/features_1_6765d2890b.png 772w",
            sizes: "(min-width: 772px) 772px, 100vw",
          },
          sources: [],
        },
        width: 772,
        height: 354,
      },
    },
  },
};

describe("Test if the RainFeatureCard component is working as expected", () => {
  it("<RainFeatureCard /> matches snapshot", () => {
    const component = render(
      <RainFeatureCard
        heading={"Create your own dashboards with your favourite visualization"}
        subText={
          "Visualizations ranging from a simple bar chart to map visualizations. User has the freedom to create multiple dashboards with different collections of visulizations."
        }
        img={imgData}
        isImage={false}
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<RainFeatureCard /> renders without exploding", () => {
    const component = render(
      <RainFeatureCard
        heading={"Create your own dashboards with your favourite visualization"}
        subText={
          "Visualizations ranging from a simple bar chart to map visualizations. User has the freedom to create multiple dashboards with different collections of visulizations."
        }
        img={imgData}
        isImage={false}
      />
    );
    expect(component.getByTestId("rain-feature-card")).toBeInTheDocument();
  });
});
