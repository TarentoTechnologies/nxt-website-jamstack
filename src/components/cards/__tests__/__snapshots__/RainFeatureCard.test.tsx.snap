// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the RainFeatureCard component is working as expected <RainFeatureCard /> matches snapshot 1`] = `
<div>
  <div
    data-testid="rain-feature-card"
  >
    <div
      class="imgStyles"
    >
      <div
        class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
        data-gatsby-image-wrapper=""
        style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
      >
        <div
          style="max-width: 772px; display: block;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;charset=utf-8,%3Csvg%20height='354'%20width='772'%20xmlns='http://www.w3.org/2000/svg'%20version='1.1'%3E%3C/svg%3E"
            style="max-width: 100%; display: block; position: static;"
          />
        </div>
        <img
          alt=""
          aria-hidden="true"
          data-placeholder-image=""
          decoding="async"
          src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAJABQDASIAAhEBAxEB/8QAFwAAAwEAAAAAAAAAAAAAAAAAAAIDBf/EABUBAQEAAAAAAAAAAAAAAAAAAAAB/9oADAMBAAIQAxAAAAHLWq1MoH//xAAZEAACAwEAAAAAAAAAAAAAAAAAEQEQITH/2gAIAQEAAQUCWaIjlf/EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQMBAT8BP//EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQIBAT8BP//EABQQAQAAAAAAAAAAAAAAAAAAACD/2gAIAQEABj8CX//EABsQAAEEAwAAAAAAAAAAAAAAAAABEBFxMUFR/9oACAEBAAE/IdiTNC2JcMNP/9oADAMBAAIAAwAAABAwD//EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQMBAT8QP//EABcRAQADAAAAAAAAAAAAAAAAAAABESH/2gAIAQIBAT8QnVP/xAAcEAEAAQQDAAAAAAAAAAAAAAABEQAQITFhkbH/2gAIAQEAAT8QJlFDUvaDwnauVbHdv//Z"
          style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
        />
        <img
          alt=""
          data-aos="fade-up"
          data-gatsby-image-ssr=""
          data-main-image=""
          data-src="/static/c5ab777b81ede6691c6b6d65988b39cb/02c3d/features_1_6765d2890b.png"
          data-srcset="/static/c5ab777b81ede6691c6b6d65988b39cb/07064/features_1_6765d2890b.png 193w,/static/c5ab777b81ede6691c6b6d65988b39cb/35e06/features_1_6765d2890b.png 386w,/static/c5ab777b81ede6691c6b6d65988b39cb/02c3d/features_1_6765d2890b.png 772w"
          decoding="async"
          loading="lazy"
          sizes="(min-width: 772px) 772px, 100vw"
          style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
        />
        <noscript />
        <script
          type="module"
        >
          const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}
        </script>
      </div>
    </div>
    <div
      class="headingStyles cardHeader carbon"
      data-aos="fade-up"
    >
      Create your own dashboards with your favourite visualization
    </div>
    <p
      class="subTextStyles carbon"
      data-aos="fade-up"
    >
      Visualizations ranging from a simple bar chart to map visualizations. User has the freedom to create multiple dashboards with different collections of visulizations.
    </p>
  </div>
</div>
`;
